sections/futures/Trade/NextPriceConfirmationModal.tsx

```diff
@@ -0,0 +1,196 @@
+import { useRecoilValue } from 'recoil';
+import styled from 'styled-components';
+import { useTranslation } from 'react-i18next';
+import Wei from '@synthetixio/wei';
+import useSynthetixQueries from '@synthetixio/queries';
+
+import { newGetExchangeRatesForCurrencies } from 'utils/currencies';
+import BaseModal from 'components/BaseModal';
+import { gasSpeedState } from 'store/wallet';
+
+import { FlexDivCol, FlexDivCentered } from 'styles/common';
+import Button from 'components/Button';
+import { newGetTransactionPrice } from 'utils/network';
+
+import GasPriceSelect from 'sections/shared/components/GasPriceSelect';
+import useSelectedPriceCurrency from 'hooks/useSelectedPriceCurrency';
+import { Synths } from 'constants/currency';
+import Connector from 'containers/Connector';
+import { zeroBN, formatCurrency } from 'utils/formatters/number';
+import { PositionSide } from '../types';
+import { GasLimitEstimate } from 'constants/network';
+import useGetNextPriceDetails from 'queries/futures/useGetNextPriceDetails';
+
+type NextPriceConfirmationModalProps = {
+	onDismiss: () => void;
+	market: string | null;
+	tradeSize: string;
+	gasLimit: GasLimitEstimate;
+	onConfirmOrder: () => void;
+	side: PositionSide;
+	l1Fee: Wei | null;
+	feeCost: Wei | null;
+	positionSize: Wei | null;
+};
+
+const NextPriceConfirmationModal: FC<NextPriceConfirmationModalProps> = ({
+	onDismiss,
+	market,
+	tradeSize,
+	side,
+	gasLimit,
+	onConfirmOrder,
+	l1Fee,
+	feeCost,
+	positionSize,
+}) => {
+	const { t } = useTranslation();
+	const { synthsMap } = Connector.useContainer();
+	const gasSpeed = useRecoilValue(gasSpeedState);
+	const { useExchangeRatesQuery, useEthGasPriceQuery } = useSynthetixQueries();
+	const { selectedPriceCurrency } = useSelectedPriceCurrency();
+	const ethGasPriceQuery = useEthGasPriceQuery();
+	const exchangeRatesQuery = useExchangeRatesQuery();
+	const nextPriceDetailsQuery = useGetNextPriceDetails(market);
+
+	const gasPrices = useMemo(
+		() => (ethGasPriceQuery.isSuccess ? ethGasPriceQuery?.data ?? undefined : undefined),
+		[ethGasPriceQuery.isSuccess, ethGasPriceQuery.data]
+	);
+
+	const exchangeRates = useMemo(
+		() => (exchangeRatesQuery.isSuccess ? exchangeRatesQuery.data ?? null : null),
+		[exchangeRatesQuery.isSuccess, exchangeRatesQuery.data]
+	);
+
+	const ethPriceRate = useMemo(
+		() => newGetExchangeRatesForCurrencies(exchangeRates, Synths.sETH, selectedPriceCurrency.name),
+		[exchangeRates, selectedPriceCurrency.name]
+	);
+
+	const gasPrice = ethGasPriceQuery.data != null ? ethGasPriceQuery.data[gasSpeed] : null;
+	const nextPriceDetails = nextPriceDetailsQuery?.data;
+
+	const transactionFee = useMemo(
+		() => newGetTransactionPrice(gasPrice, gasLimit, ethPriceRate, l1Fee),
+		[gasPrice, gasLimit, ethPriceRate, l1Fee]
+	);
+
+	const orderDetails = useMemo(() => {
+		const newSize = side === PositionSide.LONG ? tradeSize : -tradeSize;
+
+		return { size: (positionSize ?? zeroBN).add(newSize) };
```

Comment:
This logic isn't correct. If I have a short position open and I'm increasing the short, it subtracts `newSize` from `positionSize`, but `positionSize` has already applied `.abs()`

Example: I'm short 1 ETH. I submit an order to increase my short by .5 ETH. The modal will display `Size: .5 ETH`